### DISCUSSION TITLE
How to delete repositories and artifacts from Harbor database when files don't exist anymore

### DISCUSSION THREAD

**Discussion by kuyajon** [goharbor/harbor#20095] [General]
We have a unique setup where we use Harbor as a mirror. So we setup a registry and replication, so we could pull from another repository.

But since we are just pulling, we don't need to backup the files of the artifacts. We just want to backup the setup, and if we need to restore, we can just pull again.

I tried to backup the database, redis, etc. And then I reinstalled a fresh Harbor, and restored those data. It worked! My problem is, it thinks the instance still have the binaries of all the data it pulled before. I don't know how to clean this up. If I execute the replication, it does not pull anymore because harbor thinks we have the latest data.

Any ideas what I can do? Maybe execute some sql commands? that sounds too intrusive.

**Comment by kuyajon** [comment]
I was able to resolve by deleting contents of the following tables in order:     "tag"
    "task"
    "artifact_trash"
    "artifact_accessory"
    "artifact_blob"
    "artifact_reference"
    "artifact"
    "blob"
    "project_blob"
    "repository"
    "execution"

**Reply by FuxMak** [reply]
Unfortunately I've ran into the same issue and clearing all entries related to the digest did not resolve the issue.

Any chance you documented the commands you were using?
